src/sentry/users/api/serializers/authenticator.py

```diff
@@ -1,15 +1,55 @@
+from __future__ import annotations
+
+from collections.abc import Mapping
+from datetime import datetime
+from typing import TYPE_CHECKING, Any, TypedDict, cast
+
 from sentry.api.serializers import Serializer, register
 from sentry.auth.authenticators.base import AuthenticatorInterface
 from sentry.auth.authenticators.recovery_code import RecoveryCodeInterface
 from sentry.auth.authenticators.sms import SmsInterface
 from sentry.auth.authenticators.totp import TotpInterface
 from sentry.auth.authenticators.u2f import U2fInterface
 
+if TYPE_CHECKING:
+    from django.utils.functional import _StrOrPromise
```

Comment:
From: https://github.com/typeddjango/django-stubs/issues/1804
is it ok if we potentially return/serialize a promise?

Reply:
this is actually fine -- `_StrOrPromise` is ~kinda public and typing-only